tests/std/tests/P0896R4_P1614R2_spaceship/test.cpp

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// Covers:
+// * spaceship for containers
+
+#include <array>
+#include <cassert>
+#include <compare>
+#include <concepts>
+#include <deque>
+#include <forward_list>
+#include <functional>
+#include <iostream>
+#include <list>
+#include <map>
+#include <queue>
+#include <ranges>
+#include <set>
+#include <stack>
+#include <string>
+#include <type_traits>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+template <class Container>
+void ordered_containers_test(const Container& smaller, const Container& smaller_equal, const Container& larger) {
+    assert(smaller < larger);
+    assert(smaller <= larger);
+    assert(larger > smaller);
+    assert(larger >= smaller);
+    assert(smaller == smaller_equal);
+    assert(smaller != larger);
+    assert((smaller <=> larger) < 0);
+    assert((larger <=> smaller) > 0);
+    assert((smaller_equal <=> smaller) == 0);

```

Comment:
Comparing with `0` tells us that `<=>` is returning the right value, but it doesn't tell us if it's returning the right _type_. Since all of the callers are passing in containers of strongly-ordered types, `static_assert(std::is_same_v<decltype(smaller <=> smaller), std::strong_ordering>)` would validate that.

Reply:
Interestingly, because `pair` doesn't yet support spaceship, `_Synth_three_way` activates and returns `weak_ordering`. I've implemented a TRANSITION workaround, so that when `pair` enters the Space Age, the test will fail and we'll be reminded to remove the workaround.